loopy/target/c/__init__.py

```diff
@@ -899,14 +899,15 @@ def get_temporary_decls(self, codegen_state, schedule_index):
 
         ecm = self.get_expression_to_code_mapper(codegen_state)
 
-        for tv in sorted(
-                kernel.temporary_variables.values(),
-                key=lambda key_tv: key_tv.name):
+        for tv_name in sorted(kernel.temporary_variables):
+            tv = kernel.temporary_variables[tv_name]
+            if tv.name not in sub_knl_temps:
+                continue
+
```

Comment:
```suggestion
        for tv_name in sorted(sub_knl_temps):
            tv = kernel.temporary_variables[tv_name]
```
I guess they are the same strictly better performing? (Please lmk if I'm missing something)